packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/ReactTextShadowNodeUtils.kt

```diff
@@ -0,0 +1,81 @@
+package com.datadog.reactnative.sessionreplay
+
+import android.util.Log
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.uimanager.ReactShadowNode
+import com.facebook.react.uimanager.UIImplementation
+import com.facebook.react.uimanager.UIManagerModule
+import com.facebook.react.views.text.ReactTextShadowNode
+import com.facebook.react.views.text.TextAttributes
+import okhttp3.internal.notify
+import okhttp3.internal.wait
+
+internal class ReactTextShadowNodeUtils(
+    private val reactContext: ReactContext
+) {
+    internal fun getFontSize(shadowNode: ReactTextShadowNode): Int {
+        shadowNode.javaClass.superclass?.getDeclaredField("mTextAttributes").let {
+            it?.isAccessible = true
+            val textAttributes = it?.get(shadowNode) as TextAttributes
+            return textAttributes.effectiveFontSize
+        }
+    }
+
+    internal fun getFontFamily(shadowNode: ReactTextShadowNode): String? {
+        shadowNode.javaClass.superclass.getDeclaredField("mFontFamily").let {
+            it.isAccessible = true
+            return it.get(shadowNode)?.toString()
+        }
+    }
+
+    internal fun getGravity(viewId: Int): Int? {
+        val shadowNode = getShadowNode(viewId) as? ReactTextShadowNode ?: return null
```

Comment:
Question: why do we pass the viewID instead of the shadowNode here? Accessing the shadowNode is very time-consuming since we need to jump to another thread, so it would be best to do it only once per node.

Reply:
Yes, you're right - I removed this everywhere else and it looks like I missed a place. I'll push a fix for it